syft/get_source_config.go

```diff
@@ -61,29 +59,6 @@ func (c *GetSourceConfig) WithDefaultImagePullSource(defaultImagePullSource stri
 	return c
 }
 
-func (c *GetSourceConfig) getProviders(userInput string) ([]source.Provider, error) {
-	providers := collections.TaggedValueSet[source.Provider]{}.Join(sourceproviders.All(userInput, c.SourceProviderConfig)...)
-
-	// if the "default image pull source" is set, we move this as the first pull source
-	if c.DefaultImagePullSource != "" {
-		base := providers.Remove(sourceproviders.PullTag)
-		pull := providers.Select(sourceproviders.PullTag)
-		def := pull.Select(c.DefaultImagePullSource)
-		if len(def) == 0 {
-			return nil, fmt.Errorf("invalid DefaultImagePullSource: %s; available values are: %v", c.DefaultImagePullSource, pull.Tags())
-		}
-		providers = base.Join(def...).Join(pull...)
```

Comment:
This set is supposed to remove duplicates, but I see through some debugging that some of the sources are not comparable, so they are not being removed. An easier change would just be to add this line:
`pull = pull.Remove(c.DefaultImagePullSource)`